spring-boot-project/spring-boot/src/main/java/org/springframework/boot/LazyInitializationBeanFactoryPostProcessor.java

```diff
@@ -36,8 +57,16 @@
 	@Override
 	public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory)
 			throws BeansException {
-		for (String name : beanFactory.getBeanDefinitionNames()) {
-			BeanDefinition beanDefinition = beanFactory.getBeanDefinition(name);
+
+		List<EagerLoadingBeanDefinitionPredicate> eagerPredicateList = getEagerLoadingClassesFromContext(
+				beanFactory);
+
+		for (String beanName : beanFactory.getBeanDefinitionNames()) {
+			BeanDefinition beanDefinition = beanFactory.getBeanDefinition(beanName);
+			if (eagerPredicateList.stream().anyMatch(
+					(predicate) -> predicate.test(beanFactory.getType(beanName)))) {
```

Comment:
I believe this may lead to factory beans bean instantiated so we need to avoid it. We probably need something that's somewhat similar to `BeanTypeRegistry`. Making that class public doesn't feel like a good idea, but it's also complicated enough that copying it doesn't feel like a good idea either. In short, a change is needed here but I don't know what it should be. I wonder if Framework could provide an API for getting a bean's type that gives up and returns null rather than instantiating a factory bean that lacks a generic type.

Reply:
Or we can revert back to using a customizer that passes in a list of classes that should be ignored? This basically inverts the problem and allows the use of `beanFactory.isTypeMatch()`. I really like the idea of a Predicate (as you can build much smarter ignore logic) but I am wondering if there are any cases where a list of classes wouldn't suffice?

Reply:
Unfortunately, we can't safely use `isTypeMatch()` either. If a bean is a `FactoryBean`, Framework may instantiate it to determine the type of the object that it produces.

Reply:
One thing that could be done (at least for collecting the Predicates) would be to use `SpringFactoriesLoader` rather than scanning the list of bean definitions. Of course, once we have the list of predicates we are still back to the need of passing each bean definition's type to those predicates.

Reply:
We considered `spring.factories` (it has some notable benefits over beans being created very early) but preferred the `@Bean`-based approach that allows the predicate and the auto-configuration to which it relates to be defined together in the same class.